### PR TITLE
Remember history improvements

### PR DESCRIPTION
**Pull request by cyian-1756**
# Category

This change is exactly one of the following (please change `[ ]` to `[x]`) to indicate which:
* [ ] a bug fix (Fix #...)
* [ ] a new Ripper
* [ ] a refactoring
* [ ] a style change/fix

# Description

Now by default url histories are saved in a file named after the url in $RipDir/history.txt

If the user turns on Remember global url history then the urls will be saved to $configDir/url_history.txt

# Testing

Required verification:
* [ ] I've verified that there are no regressions in `mvn test` (there are no new failures or errors).
* [ ] I've verified that this change works as intended.
  * [x] Downloads all relevant content.
  * [ ] Downloads content from multiple pages (as necessary or appropriate).
  * [ ] Saves content at reasonable file names (e.g. page titles or content IDs) to help easily browse downloaded content.
* [ ] I've verified that this change did not break existing functionality (especially in the Ripper I modified).

Optional but recommended:
* [ ] I've added a unit test to cover my change.

### CODE REVIEW

**Comment by cyian-1756** [comment]
For some reason the OracleJDK test is failing on a file I didn't change, no idea what's going on there

**Comment by metaprime** [comment]
I really think this kind of change would be better if we keep the remembered url history files in the output directory of the rip. That would definitely work better with portable mode. Maybe that should be a checkbox at a minimum, if some people would not like that.

**Comment by rautamiekka** [comment]
^ @metaprime makes a good point.

**Comment by cyian-1756** [comment]
> I really think this kind of change would be better if we keep the remembered url history files in the output directory of the rip

Do you know any way to get the current rip dir? tbh the biggest reason I went with history_files was because I couldn't figure out how to get the current rip dir

> That would definitely work better with portable mode.

In portable mode $configDir is set to the same dir as ripme.jar

**Comment by metaprime** [comment]
@cyian-1756 
> Do you know any way to get the current rip dir? 

If it's not readily available we should absolutely refactor to make that possible.

**Comment by cyian-1756** [comment]
> If it's not readily available we should absolutely refactor to make that possible.

Then we must refactor! Because I can't think of anyway to get the folder that would work on all rippers/not be a compete hack

**Comment by cyian-1756** [comment]
@metaprime What function was it that got the current rip dir? You commented it early in another thread but I closed the tab and can't find it now

**Comment by metaprime** [comment]
AbstractRipper.java

    /**
     * @return
     *      Path to the directory in which all files
     *      ripped via this ripper will be stored.
     */
    public File getWorkingDir() {
        return workingDir;
    }

**Comment by cyian-1756** [comment]
@metaprime 

> I really think this kind of change would be better if we keep the remembered url history files in the output directory of the rip.

Done in my last push.

**Comment by cyian-1756** [comment]
@metaprime You wanna look over my code again?

File: src/main/java/com/rarchives/ripme/utils/Utils.java [hunk 1]
```diff
@@ -197,14 +197,27 @@ public static String getConfigDir() {
         }
     }
     // Delete the url history file
-    public static void clearURLHistory() {
-        File file = new File(getURLHistoryFile());
-        file.delete();
+//    public static void clearURLHistory() {
+//        File file = new File(getURLHistoryFile());
+//        file.delete();
+//    }
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/utils/Utils.java]: Remove dead code instead of commenting it out.

**Comment by metaprime** [comment]
I'll need to give it a try.

**Pull request closed without merging**